### PR TITLE
Feat design implementation

### PR DESCRIPTION
**Pull request by asimpk**

### CODE REVIEW

**Comment by udosson** [comment]
@asimpk - is this the recent state of the front end deployed to the server?

Can you please list what features are included in this PR`?

**Comment by asimpk** [comment]
> @asimpk - is this the recent state of the front end deployed to the server?
> 
> Can you please list what features are included in this PR`?

hi @udosson, yes the deployed frontend is synced with this PR. 
Features or Major Changes:
SDK:
Multi-Asset support.
State Management.
Tests for multi-asset.

Frontend:
Design Update
Multi-Asset UI Components
Wallet Integration

File: packages/contracts/contracts/KumoFaucet.sol
Comment (udosson) [inline packages/contracts/contracts/KumoFaucet.sol:1]: Let's move this file to `./contracts/TestContracts`. What to you think, @arthurka-o

File: packages/dev-frontend/src/components/AddAssetButton.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
```
Comment (udosson) [inline packages/dev-frontend/src/components/AddAssetButton.tsx:1]: Is this how this should be written? Or should it be:
`import { React,  useEffect, useState } from "react";`

File: packages/dev-frontend/src/components/RiskyTroves.tsx
Comment (udosson) [inline packages/dev-frontend/src/components/RiskyTroves.tsx:1]: Can the `nbc` and `csc` parts be more generic? How would this work if we added more features? This way, we also need to add the assets hard-coded here

File: packages/dev-frontend/src/hooks/DashboardContext.tsx
Comment (udosson) [inline packages/dev-frontend/src/hooks/DashboardContext.tsx:1]: Let's make this more generic.
e.g.
totalColl[NBC] or so

File: packages/lib-ethers/src/BlockPolledKumoStore.ts [hunk 2]
```diff
@@ -155,7 +158,8 @@ export class BlockPolledKumoStore extends KumoStore<BlockPolledKumoStoreExtraSta
                 Decimal.ZERO,
                 AddressZero
               ),
-              kumoStake: new KUMOStake()
+              kumoStake: new KUMOStake(),
+              testTokensTransfered: false
```
Comment (udosson) [inline packages/lib-ethers/src/BlockPolledKumoStore.ts:162]: @asimpk - can you start thinking about a test and a production deployment? How should these be different and how to deploy these based on a given parameter? 
e.g. testTokens will only be needed for the test version and shouldn't be added to the production-ready system

**Review by udosson — CHANGES_REQUESTED** [review]
@asimpk - thanks for the PR. It already looks very good.
My comments are mainly related to the generic architecture and the differentiation between test and production-ready systems.

File: packages/dev-frontend/src/components/AddAssetButton.tsx (hunk 1 above)
Comment (asimpk) [reply packages/dev-frontend/src/components/AddAssetButton.tsx:1]: Hi, this is related to named and default exports and imports in javascript,
We have to write in {} for importing the named modules and Defaults imported as React.

File: packages/dev-frontend/src/components/AddAssetToken/context/AssetViewProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import React, { useState, useCallback, useEffect, useRef } from "react";
```
[the remark this answers is not in the document]
Comment (asimpk) [reply packages/dev-frontend/src/components/AddAssetToken/context/AssetViewProvider.tsx:1]: > Same here. Or is this they way to import `React`?

explained in above comment

**Comment by udosson** [comment]
@asimpk - the changes look good in the code. Have you tested the main function in the front end? Maybe you can try to use your E2E tests as an additional check.

**Comment by asimpk** [comment]
> @asimpk - the changes look good in the code. Have you tested the main function in the front end? Maybe you can try to use your E2E tests as an additional check.

My test work is different, and maybe needs some fixes also I want to update that branch with the latest code from the dev after this merge

**Comment by udosson** [comment]
Alright. I'm okay with that.

**Review by udosson — APPROVED** [review]

**Comment by udosson** [comment]
@arthurka-o - can you please also have a look at this PR? thx

**Review by arthurka-o — APPROVED** [review]

**Pull request merged**